example/basicauth/main.go

```diff
@@ -43,7 +44,8 @@ func main() {
 	user, _, err := client.Users.Get(ctx, "")
 
 	// Is this a two-factor auth error? If so, prompt for OTP and try again.
-	if _, ok := err.(*github.TwoFactorAuthError); ok {
+	var twoFactorAuthError *github.TwoFactorAuthError
+	if errors.As(err, &twoFactorAuthError) {
```

Comment:
When we do not need the value, I much prefer the single-line solution here and throughout this PR, which is what I was attempting to do previously (but unfortunately I got it wrong before). Now I've tried this out, and it works.  Please go ahead and update this PR so that any place we don't need the value, it transforms into a one-liner use of `errors.As`, like this:


```suggestion
	if errors.As(err, new(*github.TwoFactorAuthError)) {
```

Reply:
That's a good plan!